xbmc/cores/AudioEngine/Sinks/AESinkAUDIOTRACK.cpp

```diff
@@ -496,17 +496,25 @@ bool CAESinkAUDIOTRACK::Initialize(AEAudioFormat &format, std::string &device)
     {
       m_format.m_frameSize = m_format.m_channelLayout.Count() * (CAEUtil::DataFormatToBits(m_format.m_dataFormat) / 8);
       m_sink_frameSize = m_format.m_frameSize;
-      bool isHDiec = ((m_format.m_streamInfo.m_type == CAEStreamInfo::STREAM_TYPE_TRUEHD) ||
-                      (m_format.m_streamInfo.m_type == CAEStreamInfo::STREAM_TYPE_DTSHD_MA));
-      if (m_passthrough && isHDiec)
+
+      const bool isTrueDH = (m_format.m_streamInfo.m_type == CAEStreamInfo::STREAM_TYPE_TRUEHD);
+      const bool isDtsHD = (m_format.m_streamInfo.m_type == CAEStreamInfo::STREAM_TYPE_DTSHD_MA);
+
+      // Certain boxes have issues opening DTS-HD / TrueHD with this large amount of data
+      // adjust accordingly
+      if (m_passthrough && isDtsHD)
       {
-        // Certain boxes have issues opening DTS-HD / TrueHD with this large amount of data
-        // adjust accordingly
         m_min_buffer_size *= 2;
         m_format.m_frames = static_cast<int>(m_min_buffer_size / m_format.m_frameSize) / 2;
         m_audiotrackbuffer_sec =
             static_cast<double>(m_min_buffer_size) / (m_sink_frameSize * m_sink_sampleRate);
       }
+      else if (m_passthrough && isTrueDH)
+      {
```

Comment:
From a logic POV, this now handles RAW and IEC in the same condition, while I strongly separated it before. Reason why I did this is: In the TrueDH case, you would now miss to set the proper m_min_buffer_size for TrueHD IEC.

Reply:
Should that be `isTrueHD` instead of `isTrueDH`?

Reply:
yes typo

Reply:
@fritsch Yes probably works fine with `m_min_buffer_size *= 2;` too.

I'm not really sure that changes to this part are necessary. I changed several things at the same time since due to time constraints I couldn't do many tests... It just worked fine the first time with IEC & RAW with all the changes.

Since the audio keeps coming in groups of 24 to the sink probably the same buffer length as before is fine now. I have to do more tests.

Reply:
Tested and changes here are not necessary, removed. 

`m_min_buffer_size *= 2;` it even seems that it behaves better when skip chapters. 
Only changes need using IEC is in `CAESinkAUDIOTRACK::AddPackets`